app.js

```diff
@@ -54,37 +118,40 @@ fs.readdir('./files/', 'utf8', (error, files) => {
             } else {
                 createTable += colsName.join(" TEXT, ") + " TEXT)";
             }
+        }
 
+        if (args[2] === "--overwrite") {
 
-            connection.query(createTable, error => {
+            connection.query(`SHOW TABLES LIKE '${tableName.slice(1, -1).replace(/'/g, "\\'")}'`, (error, result) => {
                 if (error) {
                     console.error(error);
                     return;
                 }
-
-                let completedQueryCount = 0;
-                data.forEach((row, index, array) => {
-                    let rowToInsert = [];
-                    for (let col in row) {
-                        rowToInsert.push(row[col]);
-                    }
-
-                    connection.query(`INSERT INTO ${tableName} VALUES (${rowToInsert.map(x => {
-                        x = x.replace(/"/g, '\\"');
-                        return '"' + x + '"';
-                    }).join()})`, error => {
-                        // Close the connection once all queries are complete
-                        if (++completedQueryCount === array.length) {
-                            connection.end();
-                        }
-
+                if (result.length) {
```

Comment:
If the table does not exist then you are not creating it and inserting data. So if I run this program with --overwrite the first time, and the table does not exist, then the program does nothing and just hangs (because the connection is never closed either).